app/code/Magento/Swatches/Block/Product/Renderer/Configurable.php

```diff
@@ -182,6 +182,9 @@ public function getJsonSwatchConfig()
                     $attributeDataArray
                 );
             }
+            if (isset($attributeDataArray['additional_data'])) {
+            $config[$attributeId]['additional_data'] =  $attributeDataArray['additional_data'];
```

Comment:
Proper alignment makes code much easier to read

```suggestion
                $config[$attributeId]['additional_data'] = $attributeDataArray['additional_data'];
```

Reply:
Also, what do you think about assigning only 'update_product_preview_image' key?

Reply:
Hello @sivaschenko 
in additional_data we get all data in serialize form
 {swatch_input_type: "text", update_product_preview_image: "1", use_product_image_for_swatch: 0}